server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+
+    //set a daemon flag when the program is run with the -d argument
+    int daemon_flag = 0;
+
+    if(argc == 2)
+    {
+        char *daemon_argument = argv[1];
+        if(strcmp(daemon_argument, "-d") != 0)
+        {
+            printf("Invalid argument. Expected argument for daemon is \"-d\\n");
+        }
+
+        else
+        {
+            daemon_flag = 1;
+        }
+    }
+
+    signal(SIGINT, sig_handler); //Register the signal handler for SIGINT
+    signal(SIGTERM, sig_handler); //Register signal handler for SIGTERM
+
+    //Set socket parameters to stream (TCP/IP) and AF_INET for IPv4
+    sock_fd = socket(AF_INET, SOCK_STREAM, 0);
+    
+    if(sock_fd < 0)
+    {
+        printf("Error creating the socket, file descriptor not obtained. Error code: %d\n", errno);
+        return -1;
+    }
+
+    int status;
+    struct addrinfo hints;
+    struct addrinfo *servinfo; //Pointer to be passed to getaddrinfo()
+    int yes = 1;
+
+    //Initialize hints values
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+
+    //Using getaddrinfo to get the addresses
+    status = getaddrinfo(NULL, "9000", &hints, &servinfo);
+
+    if(status != 0)
+    {
+        printf("Error in getting address information. Error number is %d\n", errno);
+        return -1;
+    }
+
+    //set socket options below
+    if(setsockopt(sock_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes)) < 0)
+    {
+        printf("Error setting options for FD. Error code: %d\n", errno);
+        return -1;
+    }
+    
+    //Complete bind with the socket
+    status = bind(sock_fd, servinfo->ai_addr, servinfo->ai_addrlen);
+
+    if(status < 0)
+    {
+        printf("Error binding socket. Error code : %d\n", errno);
+        return -1;
+    }
+
+   
+    //servinfo is malloc'ed by the getaddrinfo function. This needs to be freed. 
+    freeaddrinfo(servinfo);
+
+     if(daemon_flag == 1)
+    {
+        int retval = daemon_creation();
+        if(retval == -1)
+        {
+            syslog(LOG_ERR, "Error creating a daemon");
+        }
+
+        else if(retval == 0)
+        {
+            syslog(LOG_DEBUG, "Created Daemon");
+        }
+    }
+
+
+    //listen for incoming connections. 
+    status = listen(sock_fd, BACKLOG_VAL);
+    if(status < 0)
+    {
+        printf("Error with listen(). Error code: %d\n", errno);
+        return -1;
+    }
+
+    //permissions for this file is usr=rw, g=rw, oth = r
+    filetotest_fd = open("/var/tmp/aesdsocketdata", O_RDWR | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+    if(filetotest_fd < 0)
+    {
+        syslog(LOG_ERR, "Error opening the file in this location with code : %d", errno);
+    }
+
+    //Creating receive buffer to recieve and store incoming packets
+    char buffer[1024];
+
+    //Allocate storage buffer
+    packet_buffer = (char *) malloc(1024);
+
+    int packet_buffer_len = 1024;
```

Comment:
@suja8742 use a #define or CONST INT for these copies of 1024 everywhere.

Reply:
Noted.